pandas/io/parsers/python_parser.py

```diff
@@ -555,20 +555,18 @@ def _infer_columns(
                 columns = self._handle_usecols(
                     columns, columns[0], num_original_columns
                 )
-            else:
-                if self.usecols is None or len(names) >= num_original_columns:
-                    columns = self._handle_usecols([names], names, num_original_columns)
-                    num_original_columns = len(names)
-                else:
-                    if not callable(self.usecols) and len(names) != len(self.usecols):
-                        raise ValueError(
-                            "Number of passed names did not match number of "
-                            "header fields in the file"
-                        )
-                    # Ignore output but set used columns.
-                    self._handle_usecols([names], names, ncols)
-                    columns = [names]
-                    num_original_columns = ncols
+            elif self.usecols is None or len(names) >= num_original_columns:
+                columns = self._handle_usecols([names], names, num_original_columns)
+                num_original_columns = len(names)
+            elif not callable(self.usecols) and len(names) != len(self.usecols):
+                raise ValueError(
+                    "Number of passed names did not match number of "
+                    "header fields in the file"
+                )
+            # Ignore output but set used columns.
+            self._handle_usecols([names], names, ncols)
+            columns = [names]
+            num_original_columns = ncols
```

Comment:
does this not have to be in an `else` block?

Reply:
Let me check

Reply:
thanks

also, please make sure to run `pre-commit run --all-files`, as some other hooks are erroring